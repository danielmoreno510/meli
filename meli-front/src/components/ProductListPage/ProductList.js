import { useHistory } from "react-router-dom";
import PropTypes from "prop-types";

import { formatCurrency } from "../../utils/currencyUtils";

const ProductList = ({ productList, getProduct }) => {
  const history = useHistory();
  const goToProduct = (product) => () => {
    history.push(`/items/${product.id}`);
    getProduct(product.id);
  };

  return productList ? (
    productList.map((product) => (
      <div
        key={product.id}
        className="result-product"
        data-testid="result-product"
      >
        <div className="reuslt-image-container">
          <img
            className="result-image"
            src={product.picture}
            alt={product.title}
            onClick={goToProduct(product)}
            data-testid="result-image"
          />
        </div>
        <div className="container-result">
          <div className="content-result">
            <div className="result-amount">
              {formatCurrency(product.price.currency, product.price.amount)}
            </div>
            <div className="result-city">{product.city_name}</div>
          </div>
          <h1 className="result-title">{product.title}</h1>
          <div className="result-condition">{product.condition}</div>
        </div>
      </div>
    ))
  ) : (
    <div />
  );
};

ProductList.propTypes = {
  productList: PropTypes.array,
  getProduct: PropTypes.func,
};

export default ProductList;
