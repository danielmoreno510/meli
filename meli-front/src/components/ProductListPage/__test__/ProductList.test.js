import React from "react";
import { shallow } from "enzyme";

import ProductList from "../ProductList";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useLocation: jest
      .fn()
      .mockReturnValue({ pathname: "/items", search: "?search=iphone%2012" }),
      useHistory: () => ({
        push: mockHistoryPush,
      }),
  };
});

describe("ProductListPage component", () => {
  const testProps = {
    productList: [
      {
        id: "MLA916044386",
        title: "iPhone 11 64 Gb Blanco",
        installments: { currency_id: "ARS" },
        price: { currency: "ARS", amount: "123456" },
        address: { city_name: "Mar de la plata" },
      },
    ],
    getProduct: jest.fn(),
  };

  describe("WHEN the ProductList component is render", () => {
    const wrapper = shallow(<ProductList {...testProps} />);

    it("THEN should display ProductList component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe("WHEN the Image is pressed", () => {
      it(`THEN push location should be called /items/${testProps.productList[0].id}`, () => {
        wrapper.find(`[data-testid="result-image"]`).simulate("click");
        expect(mockHistoryPush).toHaveBeenCalledWith(
          `/items/${testProps.productList[0].id}`
        );
      });
    });
  });

  describe("WHEN the ProductList component is render without props", () => {
    const wrapper = shallow(<ProductList />);

    it("THEN should display ProductList component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
