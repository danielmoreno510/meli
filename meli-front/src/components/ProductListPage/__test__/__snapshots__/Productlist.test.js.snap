// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListPage component WHEN the ProductList component is render THEN should display ProductList component 1`] = `
<div
  className="result-product"
  data-testid="result-product"
  key="MLA916044386"
>
  <div>
    <img
      alt="iPhone 11 64 Gb Blanco"
      className="result-image"
      data-testid="result-image"
      onClick={[Function]}
    />
  </div>
  <div
    className="container-result"
  >
    <h1
      className="result-title"
    >
      iPhone 11 64 Gb Blanco
    </h1>
    <div
      className="content-result"
    >
      <div
        className="result-amount"
      >
        ARSÂ 123,456.00
      </div>
      <div />
    </div>
    <div
      className="result-condition"
    />
  </div>
</div>
`;

exports[`ProductListPage component WHEN the ProductList component is render without props THEN should display ProductList component 1`] = `<div />`;
